### PR TITLE
feat(api): import/export endpoints for branding themes and layouts

### LINKED ISSUE #1193: feat(front): add Import and Export JSON actions to Portal Layout Builder
**Is your feature request related to a problem?**
Users configuring login and authentication page layouts cannot export or import their layout trees directly.

**Desired Solution:**
Enhance the Portal Layout Builder toolbar (`front/src/pages/portal-layouts/`):
- **Export Action:** Add an Export button to download or copy the raw JSON layout tree (`BuilderNode[]`).
- **Import Modal:** Add an Import button to paste or upload a `.json` file that replaces/updates the active canvas layout tree.

**Technical Guidance:**
- Frontend path: `front/src/pages/portal-layouts/ui/page-portal-layout-builder.tsx`
- Leverage `@/lib/builder-core` and `@/lib/builder-portal`.

### LINKED ISSUE #1194: feat(api): add import and export endpoints for email templates and portal layouts
**Is your feature request related to a problem?**
API clients and automation scripts cannot import or export email templates or portal layouts in bulk.

**Desired Solution:**
Add dedicated API endpoints in `api/src/application/http/`:
- `POST /realms/{realm}/email-templates/import` (supports JSON structure & MJML payload).
- `GET /realms/{realm}/email-templates/{id}/export` (returns downloadable JSON/MJML).
- `POST /realms/{realm}/portal-layouts/import` (supports layout JSON).
- `GET /realms/{realm}/portal-layouts/{id}/export` (returns downloadable layout JSON).

**Technical Guidance:**
- Integrate with `ferriskey-mail` (`MRML` validation for MJML) and `ferriskey-portal-layouts`.

### PR DESCRIPTION
**Pull request by jorisvilardell**
This pull request introduces import and export support for email templates, portal layouts, and portal themes, including backend logic, frontend APIs, and utility functions for downloading and uploading these resources. It adds the ability to parse and convert MJML email templates to and from a builder-friendly JSON structure, and ensures robust handling of file downloads with correct filenames. The changes also provide user feedback on import operations and improve CORS header exposure for downloads.

**Import/Export Functionality**

* Added backend service methods to import email templates, portal layouts, and portal themes, including new input types and logic to handle associated resources (e.g., layouts within themes) (`core/src/application/email_template/mod.rs`, `core/src/application/portal_layouts/mod.rs`, `core/src/application/portal_theme/mod.rs`). [[1]](diffhunk://#diff-e485de6d1ec41bd0c57bf88dcc393cb7ff54ac84aa3a1cf53d4ad25549699a3fR77-R86) [[2]](diffhunk://#diff-f4fda28f2f470d77b29b05ea40cb8d73e27803cb430485dd58e9fd8fa6024035R92-R101) [[3]](diffhunk://#diff-3ea921d6d22855890be6bbfcd62cc1a625b46e39d40e6cfdc6f62c9ae7d73172R116-R198)
* Implemented frontend React Query hooks for importing email templates, portal layouts, and portal themes, with success/error toasts and cache invalidation (`front/src/api/email-template.api.ts`, `front/src/api/portal-layouts.api.ts`, `front/src/api/portal-theme.api.tsx`). [[1]](diffhunk://#diff-7bbf0dd51fc05544dc88f14245850a3df966a51e543216ef3680b81242571977R61-R81) [[2]](diffhunk://#diff-e8417fbe47aefb9d093b6f6b460d92c0099750194bad69de2c2e81dfa939246fR128-R148) [[3]](diffhunk://#diff-8e2d820eca04f05da11feae4c8b44647a1e65eb09fac507f44031bed81652616R92-R111)

**MJML Email Template Parsing and Conversion**

* Added a method to parse MJML markup into a builder JSON structure, including robust error handling, and provided tests for round-tripping and edge cases (`core/src/infrastructure/email_template/renderer/mjml_renderer.rs`). [[1]](diffhunk://#diff-44f2f4667cce213b48a15b913fb2ab908dbdbdf0750a88be2aa22eb855db78b0R30-R175) [[2]](diffhunk://#diff-44f2f4667cce213b48a15b913fb2ab908dbdbdf0750a88be2aa22eb855db78b0R464-R535)
* Updated test mocks to support the new `parse_intermediate` method (`core/src/domain/trident/services.rs`).

**Frontend Download Utilities**

* Added a utility module to handle downloading exports, extracting filenames from the `Content-Disposition` header, and reading user-uploaded files as JSON (`front/src/api/builder-export.ts`).

**CORS and Header Exposure**

* Updated the backend server to expose the `Content-Disposition` header for cross-origin downloads, ensuring the frontend can access the correct filename for downloaded exports (`api/src/application/http/server/http_server.rs`). [[1]](diffhunk://#diff-3f7e1ac039637b684578ba0876f2a120aef9e59d418c1881ebc2042c758b8cf4L29-R31) [[2]](diffhunk://#diff-3f7e1ac039637b684578ba0876f2a120aef9e59d418c1881ebc2042c758b8cf4R114-R117)

These changes collectively provide a robust and user-friendly import/export experience for templates, layouts, and themes, with strong backend validation and improved interoperability between the frontend and backend.

## Summary by CodeRabbit

* **New Features**
  * Import and export email templates as JSON or MJML.
  * Import and export portal layouts and themes, including page content, design settings, and optional layouts.
  * Newly created portal themes now include default page compositions.
  * Added webhook events for web-origin and SAML configuration changes.

* **Bug Fixes**
  * Downloads now retain the server-provided filename instead of using a generic name.
  * Webhook headers are no longer incorrectly prefilled when editing, and omitted headers preserve existing values.
  * Invalid import files and portal layout structures now show clearer validation errors.

### CODE REVIEW

**Comment by jorisvilardell** [comment]
image

This feature implement "export" buttons to download themes/layouts and easily edit it manually or with an AI assistant. Furthermore you can import your edited themes/layouts with the "import" button.

File: front/src/api/api.client.ts [hunk 1]
```diff
@@ -2082,6 +2172,21 @@ export namespace Endpoints {
       500: Schemas.ApiErrorResponse;
     };
   };
+  export type get_Export_template = {
+    method: "GET";
+    path: "/realms/{realm_name}/email-templates/{template_id}/export";
+    requestFormat: "json";
+    parameters: {
+      path: { realm_name: string; template_id: string; format: "json" | "mjml" };
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply front/src/api/api.client.ts]: Fixed. The root cause was on the Rust side: `ExportTemplateQuery` was missing `#[into_params(parameter_in = Query)]`, so utoipa emitted `format` as a path parameter. Added the attribute and regenerated the client — it now carries `query: Partial<{ format: "json" | "mjml" }>`.

File: libs/ferriskey-api-core/src/api_entities/export.rs [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    /// Which builder the file belongs to: `email-template` or `portal-layout`.
+    pub ferriskey: String,
+    /// Format version, bumped when the envelope itself changes shape.
+    pub version: u32,
+    pub name: String,
+    /// Email templates only.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub email_type: Option<String>,
+    /// The builder tree (`BuilderNode[]`).
+    #[schema(value_type = Object)]
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply libs/ferriskey-api-core/src/api_entities/export.rs]: Fixed — `value_type = Vec<Object>` on both trees (the envelope's and the theme layout's), so clients type them as arrays.

File: libs/ferriskey-api-core/src/api_entities/export.rs [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+    #[test]
+    fn a_download_names_its_file_after_the_envelope() {
+        let response =
+            ExportEnvelope::portal_layout("Reset password".to_string(), serde_json::json!([]))
+                .into_download()
+                .expect("serialize");
+
+        assert_eq!(
+            response.headers().get(header::CONTENT_DISPOSITION).unwrap(),
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply libs/ferriskey-api-core/src/api_entities/export.rs]: Fixed for the assertions this PR adds: the header lookups and the MJML test helpers now use `expect` with a message saying what was missing. The remaining `unwrap()` calls in that file predate this branch and are left alone.

File: libs/ferriskey-api-email-template/src/handlers/export_template.rs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    summary = "Export email template",
+    description = "Returns the template as a downloadable file: the builder JSON envelope (`format=json`, \
+                   the default) or the rendered MJML markup (`format=mjml`).",
+    params(
+        ("realm_name" = String, Path, description = "Name of the realm"),
+        ("template_id" = Uuid, Path, description = "Email template ID"),
+        ExportTemplateQuery,
+    ),
+    responses(
+        (status = 200, description = "Email template exported successfully", content_type = "application/json", body = ExportEnvelope),
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply libs/ferriskey-api-email-template/src/handlers/export_template.rs]: Fixed. The 200 response now declares both media types via `content(...)`, so an MJML export is no longer described as returning the JSON envelope.

File: libs/ferriskey-api-portal-theme/src/handlers/export_theme.rs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            .service
+            .get_layout(
+                identity,
+                GetLayoutInput {
+                    realm_name,
+                    layout_id,
+                },
+            )
+            .await
+            .ok()
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply libs/ferriskey-api-portal-theme/src/handlers/export_theme.rs]: Fixed. Only `CoreError::NotFound` maps to `None` now; every other error is returned as an `ApiError`. You are right that swallowing them would have produced a file quietly missing its layout after a permission denial.

File: libs/ferriskey-api-portal-theme/src/handlers/import_theme.rs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    ValidateJson(payload): ValidateJson<ImportPortalThemeValidator>,
+) -> Result<Response<ImportPortalThemeResponse>, ApiError> {
+    ensure_importable(
+        payload.ferriskey.as_deref(),
+        payload.version,
+        ThemeExportEnvelope::KIND,
+    )?;
+
+    let config = serde_json::from_value(payload.config)
+        .map_err(|e| ApiError::BadRequest(format!("invalid theme config: {e}").into()))?;
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply libs/ferriskey-api-portal-theme/src/handlers/import_theme.rs]: Fixed — `config` falls back to `PortalThemeConfig::default()` when the field is absent, instead of failing on the `Null` that `serde(default)` produces for a `Value`.

File: libs/ferriskey-api-portal-theme/src/validators.rs [hunk 7]
```diff
@@ -19,6 +19,44 @@ pub struct CreateThemeValidator {
...
+    pub config: serde_json::Value,
+
+    /// One builder tree per page, keyed by page type.
+    #[serde(default)]
+    #[schema(value_type = Object)]
+    pub pages: serde_json::Value,
+
+    /// The layout carried inside the file, recreated on import.
+    #[serde(default)]
+    pub layout: Option<ImportPortalThemeLayoutValidator>,
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply libs/ferriskey-api-portal-theme/src/validators.rs:50]: Fixed — `#[validate(nested)]` added, so the layout name is held to its length constraint.

File: front/src/pages/realm/feature/page-realm-settings-edit-webhook-feature.tsx [hunk 8]
```diff
@@ -32,7 +32,11 @@ export default function PageRealmSettingsEditWebhookFeature() {
         description: webhook.description ?? '',
         endpoint: webhook.endpoint,
         subscribers: initialTriggers,
-        headers: Object.entries(webhook.headers).map(([key, value]) => ({ key, value })),
+        // The API never returns a webhook's headers — they are marked
+        // `skip_serializing` alongside the delivery secret — so there is
+        // nothing to prefill here. Submitting the form sends whatever the user
+        // enters, which replaces the stored headers.
+        headers: [],
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply front/src/pages/realm/feature/page-realm-settings-edit-webhook-feature.tsx:41]: Fixed, and thank you — this one was a real data-loss risk that I had only flagged in the commit message.

`headers` is now optional end to end. The update validator takes `Option<HashMap<..>>`, the service only rejects reserved headers when a set is actually supplied, and the repository leaves the column as `NotSet` when the caller says nothing — so editing an endpoint no longer clears the headers that authenticate deliveries. The form sends them only when the user typed some, since it cannot display what the API never returns.

File: front/src/api/builder-export.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Downloading an export cannot go through the generated client: it hands back a
+ * parsed body, while a download needs the bytes and the filename the server
+ * chose in `Content-Disposition`. So this issues the request itself, with the
+ * same bearer token the generated fetcher uses.
+ */
+async function fetchExport(path: string): Promise<{ blob: Blob; filename: string }> {
+  const token = authStore.getState().accessToken
+  const response = await fetch(new URL(path, window.apiUrl).toString(), {
+    headers: token ? { Authorization: `Bearer ${token}` } : undefined,
```
[the remark this answers is not in the document]
Comment (jorisvilardell) [reply front/src/api/builder-export.ts:12]: Not changing this here.

`window.apiUrl` is what the whole console already talks to, through the generated fetcher, and it is `http://localhost:3333` in development — the export path is not doing anything the rest of the app does not. Refusing `http:` in this one function would break local development while leaving every other authenticated call untouched.

Worth doing as a deployment-wide rule (reject non-HTTPS API URLs at bootstrap, with an explicit local-development allowance), which is a different change from this PR.

**Comment by jorisvilardell** [comment]
Tracked by #1194.

File: core/src/application/portal_theme/mod.rs [hunk 10]
```diff
@@ -112,3 +113,86 @@ impl PortalThemeService for ApplicationService {
         self.portal_theme_service.get_active_theme(input).await
     }
 }
+
+/// Importing a theme spans two domains — the theme and the layout it is framed
+/// by — so it is composed here rather than inside either service: the theme
+/// service has no layout repository, and giving it one to serve an import
+/// would widen it for every other caller.
+impl ApplicationService {
+    pub async fn import_portal_theme(
```
Comment (NathaelB) [inline core/src/application/portal_theme/mod.rs:122]: The logic must be written in the domain layer and in the application layer

**Review by NathaelB — CHANGES_REQUESTED** [review]

File: core/src/application/portal_theme/mod.rs (hunk 10 above)
Comment (jorisvilardell) [reply core/src/application/portal_theme/mod.rs:122]: Move into `libs/ferriskey-portal-theme`

**Review by NathaelB — APPROVED** [review]
